pkg/resources/data.go

```diff
@@ -549,6 +550,29 @@ func (d *TemplateData) KCMCloudControllersDeactivated() bool {
 	return false
 }
 
+// KCMInTreeCloudProviderDisabled checks is the in-tree cloud provider disabled
+// on the KCM deployment. We don't care is the roll-out complete in this case.
+func (d *TemplateData) KCMInTreeCloudProviderDisabled() bool {
+	kcm := appsv1.Deployment{}
+	if err := d.client.Get(d.ctx, ctrlruntimeclient.ObjectKey{Name: ControllerManagerDeploymentName, Namespace: d.cluster.Status.NamespaceName}, &kcm); err != nil {
```

Comment:
If there is an error when retrieving the deployment we consider the KCM as enabled? this could cause some flip behaviors. Having an explicit annotation or feature gate is more robust on that respect

Reply:
This can be mitigated by adding some retry logic. I'll look into this.

Reply:
Yes, you can check if you have an error because the object is missing or add a retry with backoff, I still think it was better to explicitly flag the cluster explicitly when CCM migration is required, but it's your call.